src/sfctl/custom_chaos_schedule.py

```diff
@@ -0,0 +1,181 @@
+
+    return TimeOfDay(hour, minute)
+
+def parse_time_range(time_range):
+    """Parse a TimeRange from string"""
+    from azure.servicefabric.models.time_range import (
+        TimeRange
+    )
+
+    if time_range is None:
+        return None
+
+    start_time = parse_time_of_day(time_range.get("StartTime"))
+    end_time = parse_time_of_day(time_range.get("EndTime"))
+
+    return TimeRange(start_time, end_time)
+
+def parse_active_time_ranges(time_ranges):
+    """Parse a list of TimeRanges from string"""
+
+    if time_ranges is None:
+        return list()
+
+    parsed_times = list()
+
+    for time_range in time_ranges:
+        parsed_times.append(parse_time_range(time_range))
+
+    return parsed_times
+
+def parse_active_days(active_days):
+    """Parse a ChaosJobActiveDays from string"""
+    #pylint: disable=line-too-long
+    from azure.servicefabric.models.chaos_schedule_job_active_days_of_week import (
+        ChaosScheduleJobActiveDaysOfWeek
+    )
+
+    if active_days is None:
+        return None
+
+    sunday = active_days.get("Sunday", False)
+    monday = active_days.get("Monday", False)
+    tuesday = active_days.get("Tuesday", False)
+    wednesday = active_days.get("Wednesday", False)
+    thursday = active_days.get("Thursday", False)
+    friday = active_days.get("Friday", False)
+    saturday = active_days.get("Saturday", False)
+
+    return ChaosScheduleJobActiveDaysOfWeek(
+        sunday, monday, tuesday, wednesday,
+        thursday, friday, saturday)
+
+def parse_job(job):
+    """Parse a ChaosJob from string"""
+    from azure.servicefabric.models.chaos_schedule_job import (
+        ChaosScheduleJob
+    )
+
+    if job is None:
+        return None
+
+    chaos_parameters = job.get('ChaosParameters')
+    active_days = parse_active_days(job.get('Days'))
+    times = parse_active_time_ranges(job.get('Times'))
+
+    return ChaosScheduleJob(chaos_parameters, active_days, times)
+
+def parse_jobs(jobs):
+    """Parse a list of ChaosJobs from string"""
+
+    if jobs is None:
+        return list()
+
+    parsed_jobs = list()
+
+    for job in jobs:
+        parsed_jobs.append(parse_job(job))
+
+    return parsed_jobs
+
+def parse_chaos_params_dictionary(chaos_parameters_dictionary):
+    """Parse a list of ChaosParameters dictionary input from string"""
+
+    from azure.servicefabric.models.chaos_parameters_dictionary_item import (
+        ChaosParametersDictionaryItem
+    )
+
+    from sfctl.custom_chaos import (
+        parse_chaos_parameters
+    )
+
+    if chaos_parameters_dictionary is None:
+        return list()
+
+    parsed_dictionary = list()
+
+    for dictionary_entry in chaos_parameters_dictionary:
+        key = dictionary_entry.get("Key")
+        value = parse_chaos_parameters(dictionary_entry.get("Value"))
+
+        parsed_dictionary.append(ChaosParametersDictionaryItem(key, value))
+
+    return parsed_dictionary
+
+#pylint: disable=redefined-builtin
```

Comment:
what builtin? I would avoid this, I think its just a naming conflict.

Reply:
set is https://docs.python.org/2/library/sets.html
I'm open to alternative command names if you have any.

Reply:
you can rename the function set_schedule or something similar to avoid the conflict